### ISSUE TITLE
GLFWError #65544 Happen, The GLFW library is not initialized

### ISSUE THREAD

**Issue by tankorsmash** [c0i/imguix#18]
I'm trying to instantiate the `cocos2d::IMGUIGLViewImpl::create("name")`, but it's returning nullptr, and its because `glGetString(GL_VERSION)` is returning null, which is failing the version check.

If I use the standard one, `cocos2d::GLViewImpl::create("name")`, it works well. After investigating, it seems like I can't do anything relating to glfw within my own project, but it seems alright in cocos2d's project. I'd imagine if I put the `IMGUIGLViewImpl` into cocos' project instead of mine, it would work, but I'd like to figure out why it's not possible this way.

I'm using cocos2d-x 3.14, and Visual Studio 2013 on Win10

**Comment by yinjimmy** [comment]
Hi tank,
Thanks for trying imguix.

I think you can upgrade the `IMGUIGLViewImpl` to cocos2d-x 3.14 version.
May I ask, how are you going to use imguix ?

**Comment by tankorsmash** [comment]
Hey, I took another swing at getting this going, using cocos2d-x 3.16.

Part of my problem the first time around was that I (in order to use some helpers I added to my local DesktopImpl) incorrectly changed `IMGUIGLViewImpl` so subclassed from `GlViewImpl`, instead of `GLView`. Changing that back fixed the issues reported in the OP, but then I ran into other issues where the hello world code wasn't working

```                
                    CCIMGUI::getInstance()->addImGUI([=](){
                    {
                        ImGui::Text("Hello, world!");
                    })
```

wouldn't get called, and that was because `GlViewImpl` was  coming through as the Desktop Impl instead, so I changed it to use the `IMGUIGLViewImpl` directly instead, and that worked:

```	
    // init imgui
    auto glview = Director::getInstance()->getOpenGLView();
    auto window = static_cast<IMGUIGLViewImpl*>(glview)->getWindow();
	CCIMGUI::getInstance()->setWindow(window);
```

tl;dr there is something on my end messing with how cocos is defining the GLViewImpl and how IMGUIX was expecting it to be. I don't think there's anything to be done except more research on my end, but I wanted to update the issue all the same.

**Comment by tankorsmash** [comment]
Trying again to get cocos2dx and glfw3 working within my own project so I can get this imguix project working. The glfw calls work within the `libcocos2d.dll` project, but when it comes back to `my_game.exe`'s context, the exact same `glfwGetWin32Window(_mainWindow)`  call fails only when called from `my_game.exe`.

Again, no solution but updating in case anyone else can recognize this issue and offer a solution.

**Comment by yinjimmy** [comment]
Hei,
I updated `ImGuiX` with cocos2d-x 3.17 and imgui 1.7 version with this [commit](https://github.com/c0i/imguix/commit/adf1123ffb81bf20a952f75492b2328f8d935279).

**Comment by tankorsmash** [comment]
I was able to get this resolved on my end by changing the linking in CC_PLATFORM_DEFINES from glfw3.lib to glfw3dll.lib and adding a define. It wasn't on your end at all, thank you!
